pyo3-ffi/src/cpython/abstract_.rs

```diff
@@ -42,14 +42,15 @@ extern "C" {
 }
 
 #[cfg(Py_3_8)]
-const PY_VECTORCALL_ARGUMENTS_OFFSET: Py_ssize_t =
-    1 << (8 * std::mem::size_of::<Py_ssize_t>() as Py_ssize_t - 1);
+const PY_VECTORCALL_ARGUMENTS_OFFSET: size_t =
+    1 << (8 * std::mem::size_of::<size_t>() as size_t - 1);
 
 #[cfg(Py_3_8)]
 #[inline(always)]
 pub unsafe fn PyVectorcall_NARGS(n: size_t) -> Py_ssize_t {
+    let n = n & !PY_VECTORCALL_ARGUMENTS_OFFSET;
     assert!(n <= (PY_SSIZE_T_MAX as size_t));
```

Comment:
Should we also remove the assertion? Python headers don't seem to have it, and with this implementation it now seems that this assertion will never be triggered due to the masking.

Reply:
I guess the suggested `.try_into().unwrap()` form would also remove the need for the assertion due to the introduction of the checked cast.

Reply:
The assertion would turn into the `unwrap`, but the runtime check could still be present. Personally, I would prefer the checked cast and let the compiler figure out that the masking may turn the check into no-op.

Reply:
Agreed